api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -221,23 +257,43 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 		AllAssignedDeployed: false,
 	}
 
-	if s.VendorContext != nil && s.VendorContext.EvaluationContext != nil {
-		context := s.VendorContext.EvaluationContext.Clone()
-		context.DeploymentSpec = deployment
-		context.Value = deployment
-		context.Component = ""
-		context.Namespace = namespace
-		deployment, err = api_utils.EvaluateDeployment(*context)
+	deploymentType := DeploymentType_Update
+	if remove {
+		deploymentType = DeploymentType_Delete
 	}
+	summary.IsRemoval = remove
 
-	if err != nil {
-		if remove {
-			log.Infof(" M (Solution): skipped failure to evaluate deployment spec: %+v", err)
-		} else {
-			summary.SummaryMessage = "failed to evaluate deployment spec: " + err.Error()
-			log.Errorf(" M (Solution): failed to evaluate deployment spec: %+v", err)
-			s.saveSummary(iCtx, deployment, summary, namespace)
-			return summary, err
+	s.saveSummaryProgress(ctx, deployment, summary, namespace)
+	defer func() {
+		s.concludeSummary(ctx, deployment, summary, namespace)
+	}()
+
+	// get the components count for the deployment
+	componentCount := len(deployment.Solution.Spec.Components)
+	apiOperationMetrics.ApiComponentCount(
+		componentCount,
+		metrics.GetSummaryOperation,
+		metrics.GetOperationType,
+	)
+
+	if !s.evaluationDisabled {
```

Comment:
What's the use case of disabling evaluation of expressions?

Reply:
I have checked with Nonso. It is a historic reason they introduced disableEvaluation. (He shared they used to encounter some issues with Symphony parser when dealing with delimiter. However, this issue should be fixed in current codebase.) It is OK we don't port this option.